src/extensions/ExtensionManager.as

```diff
@@ -212,6 +212,10 @@ public class ExtensionManager {
 					// 'r' is reporter, 'R' is requester, and 'rR' is a reporter forced to act as a requester.
 					spec[0] = 'rR';
 				}
+				if(spec[0] == 'h') {
+					// same as above, but force to use async hat blocks
+					spec[0] = 'hH';
+				}
```

Comment:
@bsb20 This is duplicating the existing issue with rR in a saved .sb2 file not reverting to r when opened online. In usability this means the checkbox (e.g. beside tilt or distance in WeDo) is permanently lost to display the watcher on the stage (even when the sb2 file is opened online). 

Basically a saved .sb2 file can end up containing either 'r' or 'rR' and therefore this check should work both ways - i.e. r to rR if offline and rR to r when online. Same for h and hH.